src/pylhe/__init__.py

```diff
@@ -120,14 +122,36 @@ def loads():
     pass
 
 
-def readLHEInit(thefile):
+def _extract_fileobj(fileobj):
+    """
+    Checks to see if a file is compressed, and if so, extract it with gzip
+    so that the uncompressed file can be returned.
+
+    Args:
+        fileobj: A file object that can be a file, file object, or a
+          compressed file.
+
+    Returns:
+        pathlib.PosixPath or gzip.GzipFile: An uncompressed file object.
+    """
+    with open(fileobj, "rb") as gzip_file:
+        header = gzip_file.read(2)
+    gzip_magic_number = b"\x1f\x8b"
+
+    if header == gzip_magic_number:
+        fileobj = gzip.GzipFile(fileobj)
+    return fileobj
+
+
+def readLHEInit(fileobj):
     """
     Read the init blocks. Return dict. This encodes the weight group
     and related things according to https://arxiv.org/abs/1405.1067
     This function returns a dict.
     """
```

Comment:
add doctring .. is  fileobj a path or a literal object?

Reply:
@lukasheinrich Can be either a path or a Python file object, I believe.

Reply:
hm I would've thought the `open()` call in `_extract_fileobj` would forcce a path argument

Reply:
We got rid of the `open` though so it is now just

```python
return gzip.GzipFile(fileobj) if header == gzip_magic_number else fileobj
```

You can see this more clearly in the tests that I added in https://github.com/scikit-hep/pylhe/pull/74/commits/13a60a10b9e747a6dcea313b99a39c508c614818

Reply:
oh oh you mean the open in the check.

Reply:
@lukasheinrich sorry I think I understand what you mean now &mdash; I was missing it before.